src/Security/Security.php

```diff
@@ -570,6 +596,32 @@ protected function preLogin()
     }
 
     /**
+     * Perform pre-logout checking and prepare a response if available prior to logout
+     *
+     * @return HTTPResponse Substitute response object if the logout process should be curcumvented.
+     * Returns null if should proceed as normal.
+     */
+    protected function preLogout()
```

Comment:
I would like to have it moved to a central handler. And all other handling extensions should extend from that one handler. It's been on my radar to clean up and centralize it.

In short:
I do not believe pre/post methods should live in Security, but in the handler. Details will follow.